### PR TITLE
GROOVY-9855: inline string concat sonner

### PR DESCRIPTION
**Pull request by eric-milles**
https://issues.apache.org/jira/browse/GROOVY-9855

### CODE REVIEW

File: src/main/java/org/apache/groovy/ast/tools/ExpressionUtils.java [hunk 1]
```diff
@@ -309,8 +309,15 @@ public static Expression transformInlineConstants(final Expression exp) {
             }
         } else if (exp instanceof BinaryExpression) {
             BinaryExpression be = (BinaryExpression) exp;
-            be.setLeftExpression(transformInlineConstants(be.getLeftExpression()));
-            be.setRightExpression(transformInlineConstants(be.getRightExpression()));
+            Expression lhs = transformInlineConstants(be.getLeftExpression());
+            Expression rhs = transformInlineConstants(be.getRightExpression());
+            if (lhs instanceof ConstantExpression && rhs instanceof ConstantExpression &&
```
Comment (danielsun1106) [inline src/main/java/org/apache/groovy/ast/tools/ExpressionUtils.java:314]: We should check `be.operation.type == Types.PLUS` too?

Comment (eric-milles) [reply src/main/java/org/apache/groovy/ast/tools/ExpressionUtils.java:314]: Yes, thanks.  I did have that when the code was located elsewhere.  I will get it updated.

**Comment by danielsun1106** [comment]
+1

**Comment by danielsun1106** [comment]
Merged. Thanks!

**Pull request merged**